### PR TITLE
SIMPLY-3153 Added support for books with content-type of application/octet-stream

### PR DESCRIPTION
**Pull request by vladimirfedorov**
**What's this do?**
Adds support for books with `Content-type` of `application/octet-stream`.

**Why are we doing this? (w/ JIRA link if applicable)**
Columbia University Libraries - Demo books fail to download because of that unsupported content type.
https://jira.nypl.org/browse/SIMPLY-3153

**How should this be tested? / Do these changes have associated tests?**
Open Columbia University Libraries - Demo library and download "Mathematical Theory of Advanced Computing"

**Dependencies for merging? Releasing to production?**
No

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
@vladimirfedorov

### CODE REVIEW

**Comment by ettore** [comment]
I don't see any problems with these changes, but did you verify with Leonard if this content-type is officially supported?

**Comment by vladimirfedorov** [comment]
Yes, we have a thread in # mobile-development "Thread about Columbia University Libraries - Demo Content-type for books", Android and server scripts support that content-type value

File: Simplified/NYPLBookAcquisitionPath.h [hunk 1]
```diff
@@ -12,6 +12,7 @@ static NSString * const _Nonnull ContentTypeFindaway = @"application/vnd.library
 static NSString * const _Nonnull ContentTypeOpenAccessAudiobook = @"application/audiobook+json";
 static NSString * const _Nonnull ContentTypeOpenAccessPDF = @"application/pdf";
 static NSString * const _Nonnull ContentTypeFeedbooksAudiobook = @"application/audiobook+json;profile=\"http://www.feedbooks.com/audiobooks/access-restriction\"";
+static NSString * const _Nonnull ContentTypeOctetStream = @"application/octet-stream";
```
Comment (ettore) [inline Simplified/NYPLBookAcquisitionPath.h:15]: can you move all these definitions to the `.m`?  It's wrong to have static variables in a header file.

Comment (vladimirfedorov) [reply Simplified/NYPLBookAcquisitionPath.h]: sure, done

**Review by ettore — APPROVED** [review]

**Pull request merged**